### PR TITLE
Issue #4871: CrowdNav_Prediction_AttnGraph feasibility smoke (cheap-lane worker)

### LINKED ISSUE #4871: feature-smoke: CrowdNav_Prediction_AttnGraph external learned baseline — license/checkpoint/transfer go-no-go
## Ask

Feasibility smoke for reusing https://github.com/Shuijing725/CrowdNav_Prediction_AttnGraph (intention-aware crowd navigation with attention-based interaction graph + trajectory prediction, ICRA 2023) as an external LEARNED baseline planner. Go/no-go data only — NO campaign, NO roster addition, NO retraining commitment in this issue.

This is the learned-class counterpart to the SICNav smoke (#4870): together they would cover both ends of the planner structural spectrum with external third-party instances.

## Verify FIRST (order matters — each step can end the issue early with a documented negative)

1. PRIOR ART CHECK: search docs/context/issue_1617_local_planner_repo_survey.md and the docs/context/ tree for any prior assessment of this repo or its CrowdNav lineage; also check whether any in-tree scaffold exists (the SICNav smoke found a dormant wrapper nobody remembered — grep robot_sf/baselines/ and algorithm_metadata.py for crowdnav/attn/graph entries). Do not rebuild what exists.
2. LICENSE: confirm the repo license permits academic benchmark reuse with public artifacts. No license file = hard stop, document and close.
3. PRETRAINED CHECKPOINTS: do the authors ship trained model weights, and under what terms? If no usable checkpoint exists, the integration cost jumps from adapter-writing to full RL retraining — document that cost class and STOP (retraining authorization is a separate maintainer decision).
4. If 1-3 pass: stage the repo pinned to a commit (scripts/tools/manage_external_repos.py pattern), build the inference dependency path only (PyTorch + their env deps; skip training extras), and run their own inference/demo path to prove the checkpoint loads and acts.
5. CONTRACT + TIMING: map their observation (robot state, human states/predictions, attention graph inputs) against robot_sf/baselines/interface.py Observation; write the thinnest adapter that produces actions on synthetic observations; measure per-step inference wall-clock (trivially fast for a NN policy, but confirm) AND — the real question — document every observation/assumption mismatch (holonomic vs differential drive, ORCA-trained humans vs social-force, sensing radius, human-count limits) with a judgment on whether zero-shot transfer into robot_sf scenarios is scientifically defensible or whether results would need a "policy evaluated out of training distribution" caveat.
6. Record the verdict in docs/benchmark_experimental_planners.md (new row): pass/fail per step above + the transfer-caveat assessment. A documented negative at ANY step is a complete, valid deliverable.

Refs #4870 (SICNav smoke, same program), #1617 (planner repo survey).

### PR DESCRIPTION
**Pull request by ll7**
## What

Feasibility smoke for reusing `Shuijing725/CrowdNav_Prediction_AttnGraph` (ICRA 2023, intention-aware crowd navigation with an attention-based interaction graph) as an external **learned** baseline planner. This is the learned-class counterpart to the SICNav smoke (#4870). Go/no-go data only — **no campaign, no roster addition, no retraining**.

**Verdict: smoke PASS (go).** All six issue gates pass; the shipped checkpoint loads and acts on synthetic Robot SF observations, and per-step inference is trivially fast.

## Why

Issue #4871 asked for an ordered go/no-go: prior-art → license → checkpoints → stage+inference → contract+timing → verdict row. Each step can end the issue early with a documented negative; none did.

## Per-gate result

| Step | Gate | Result |
| --- | --- | --- |
| 1. Prior art | No prior assessment; no dormant scaffold | **PASS.** Same author's `CrowdNav_HEIGHT` is already integrated (`crowdnav_height`) and was the template. |
| 2. License | MIT | **PASS.** |
| 3. Checkpoints | Trained RL navigation policy ships | **PASS** — no retraining. `41200.pt` ("Ours no-rand") + `41665.pt` ("Ours rand"); `00000.pt` files are the non-learning ORCA/SF baselines. |
| 4. Stage + inference | Loads and acts | **PASS.** Pinned SHA `3907731`; model-only PyTorch path (no OpenAI Baselines / `crowd_sim` / Python-RVO2 / TensorFlow). |
| 5. Contract + timing | Thinnest adapter + wall-clock | **PASS.** ~1–2 ms/step CPU, flat in neighbor count. |
| 6. Verdict row | Gate doc | **PASS.** |

## How it works

`robot_sf/planner/crowdnav_pred_attng.py` is the thinnest model-only adapter, mirroring the established `crowdnav_height.py` pattern:
- reconstructs the upstream `CrowdSimPred-v0` dict observation (5-step constant-velocity future edges + holonomic `ActionXY`) from world-frame Robot SF `Observation`,
- loads the shipped `41200.pt` attention-graph SRNN checkpoint via `torch.load(weights_only=False)` and calls `policy.act(..., deterministic=True)` directly,
- short-circuits the one heavy transitive import (`rl.networks.envs`, which otherwise drags in `gym` + Baselines + `crowd_sim`) with a minimal `VecNormalize` stub, so the network loads against PyTorch alone.

The repo is registered in `scripts/tools/manage_external_repos.py` (`uv run python scripts/tools/manage_external_repos.py stage crowdnav_pred_attng`). `tests/planner/test_crowdnav_pred_attng.py` runs the staged-checkout smoke (skips cleanly when unstaged, matching the SICNav convention).

## Validation (CPU-only, this host)

```
$ uv run pytest tests/planner/test_crowdnav_pred_attng.py -q
4 passed in 0.98s

[smoke] action toward goal (+x): vx=0.9840 vy=-0.1782
[smoke] per-step wall-clock (30-step mean after 3 warmup steps):
[smoke]   neighbors= 2: 2.226 ms/step
[smoke]   neighbors= 5: 2.215 ms/step
[smoke]   neighbors=10: 2.220 ms/step
[smoke]   neighbors=20: 2.227 ms/step
```
The deterministic action `≈ (0.98, -0.18)` for a robot at the origin with goal `(5, 0)` points strongly toward the goal — the loaded weights compute meaningful navigation, not noise. (standalone outside pytest ≈ 1.1 ms/step).

```
$ uv run pytest tests/tools/test_manage_external_repos.py -q
6 passed
$ uv run ruff check <changed files>        # All checks passed
```

## Notable finding (honest)

The bundled `trained_models/GST_predictor_non_rand/{configs/config.py,arguments.py}` **mismatches its own weights**: it declares `env_name=CrowdSimVarNum-v0` / `predict_method='none'` (spatial-edge width 2), but the `41200.pt` `state_dict` is width 12 (`base.spatial_attn.embedding_layer.0 = Linear(12→128)`), i.e. the 5-step **prediction** variant. `load_state_dict` fails under the bundled config. The adapter trusts the **state_dict** (authoritative) and uses `CrowdSimPred-v0` with constant-velocity futures (needs no TensorFlow/GST model). Documented in the smoke note so this is never re-discovered blind.

## Claim boundary (smoke evidence only)

This is a **smoke, not a roster addition**: the planner is intentionally NOT registered in `algorithm_metadata` / `algorithm_readiness` / the testing-only list, and must not enter benchmark sweeps. No campaign, no success/collision numbers for Robot SF scenarios, no retraining.

**Zero-shot transfer into Robot SF is NOT scientifically defensible** without an out-of-distribution caveat. The blockers documented in `docs/context/issue_4871_crowdnav_pred_attng_smoke.md`:
1. Holonomic `ActionXY` vs Robot SF unicycle kinematics.
2. Trained on ORCA pedestrians vs default social-force crowd (the headline OOD risk; same caveat as #4870).
3. The smoke uses constant-velocity prediction; the paper's GST-inferred prediction needs TensorFlow.
4. Sensing radius/FOV, human-count cap (20), 0.25 s control cadence, `v_pref=1.0`, open-space arena differ from Robot SF.

Roster/campaign/retraining decisions are separate maintainer calls after this smoke.

## Files

- `robot_sf/planner/crowdnav_pred_attng.py` — model-only adapter (new).
- `tests/planner/test_crowdnav_pred_attng.py` — staged-checkout smoke (new).
- `scripts/tools/manage_external_repos.py` — `crowdnav_pred_attng` RepoSpec (pinned SHA, MIT, validation command).
- `docs/context/issue_4871_crowdnav_pred_attng_smoke.md` — full contract + transfer-caveat analysis (new).
- `docs/benchmark_experimental_planners.md` — new "External learned-baseline feasibility smokes" verdict row.
- `CHANGELOG.md` — entry under Unreleased/Added.

This fully resolves issue #4871 (the smoke's required inputs all existed and were exercised; the GST/TF variant and any comparison slice are explicitly out of scope for this issue).

Closes #4871. Refs #4870, #1617.

---

This PR was produced by the ClaudeCode-harness/GLM-Coding-Pro cheap implementation lane; the review gate hardens and merges.

### CODE REVIEW

**Comment by ll7** [comment]
## Pre-gate review (GLM)

Adversarial pre-gate pass on PR #4877 (issue #4871, CrowdNav_Prediction_AttnGraph feasibility smoke). I staged the pinned external repo (`3907731`) locally and ran the real smoke end-to-end.

**Checks passed**

- **Issue coverage** — all 6 go/no-go gates addressed: prior-art grep is clean (no dormant `pred_attng`/`attngraph` scaffold in `robot_sf/` or `algorithm_metadata`); MIT; shipped `41200.pt`; stage+inference; contract+timing; verdict row. Scope is correctly smoke-only — NOT registered in `algorithm_metadata`/`algorithm_readiness`, no campaign, no retraining. `Closes #4871` is appropriate; `Refs #4870, #1617` correct.
- **Tests** — `tests/planner/test_crowdnav_pred_attng.py` → **4 passed**; they genuinely exercise the change (load the real checkpoint, assert `selfAttn_merge_SRNN` base + `nenv=1`, run inference, assert finite + v_pref-clipped action, measure per-step wall-clock, assert determinism across `reset()`). `tests/tools/test_manage_external_repos.py` → **6 passed**; the RepoSpec `validation_command` runs clean.
- **Smoke reproduces PR-body numbers** — action `(0.9840, -0.1782)` (goal-directed); ~2.2–2.3 ms/step, flat across 2/5/10/20 neighbors.
- **Determinism/robustness** — `deterministic=True`, `eval()`, `torch.no_grad()`, stable `argsort`, guarded dt compare (`math.isclose`), finite-check on dt, v_pref clip; test asserts post-reset determinism. Verified against staged upstream `selfAttn_srnn_temp_node.py`: on the `sort_humans=True` path the SRNN reads `detected_human_num` (set correctly) and does **not** read `visible_masks`, so the adapter's all-zero `visible_masks` is harmless dead data included only for obs-space shape completeness — not a masking bug.
- **Evidence hygiene** — no writes under `docs/context/evidence/`; no checksum files touched; staged checkout + provenance manifest are gitignored. Changed files = exactly the 6 declared.
- **Lint** — `ruff check` + `ruff format --check` on all changed files → clean. Full `tests/planner/` collection (590 tests) unaffected.
- **Contract mapping** — grounded in the real `robot_sf/baselines/interface.py` `Observation` (`{dt, robot, agents, obstacles}`).

**Defects found & fixed:** none. No commits pushed.

**Minor nit for the auditor (not blocking, no fix made):** the PR-body gate-5 table cell says "~1–2 ms/step", but the in-pytest figure is ~2.2 ms/step (the validation paste directly below shows 2.226/2.215/2.220/2.227, and the smoke doc records ~2.2 pytest / ~1.1 standalone). The standalone number is in range so the cell is defensible, but it slightly understates the pytest number. Doesn't move the go/no-go (trivially fast either way); the authoritative smoke doc has the correct values.

Net: **clean** from this gate.

**Comment by ll7** [comment]
Gate delta for head `92b1b842ae9cfd0658a2afb4e0b0e52bb987cf4c`: prior gate review at this SHA stands.

Review-bot thread enumeration: 0 open review threads.

No additional findings. CI is green; running the targeted local proof before merge.
